Issue 12

mcmcgrath13 opened the pull request:
* Add minification of field names
* Fix connection to work with sqlalchemy and the mysql server
* Delete unused files
* Add test for minification, combine db/parsing tests

Because the parsing of files is now more closely intertwined with the DB, the tests needed to be combined.  Also, the parsers now do reads/inserts from the DB so cannot run without a connection.

This PR closes #12, closes #13,  closes #11 and closes #18

mcmcgrath13 committed:
feat(tables): add fields for field name to code to display mapping

mcmcgrath13 committed:
feat(minify): add minification of column names to base parser and fastqc

mcmcgrath13 committed:
Merge remote-tracking branch 'origin/devel' into issue-12

mcmcgrath13 committed:
fix(connection): fix connection to work with sqlalchemy, but also the db server

mcmcgrath13 committed:
fix(minify): squash bugs, runs on fastqc now

mcmcgrath13 committed:
feat(minify): add minification of column names to qckitfastq

mcmcgrath13 committed:
test: fix existing tests

mcmcgrath13 committed:
test(minify): add test to check minification

12

mcmcgrath13 committed:
fix(db-load): re-try/except the parsing - had removed for debugging

mcmcgrath13 committed:
build(pipenv): pipenv-ify qcdb, update docker files as needed

aguang commented on qcdb/parsers/parse.py:
```diff
@@ -50,10 +57,78 @@ def get_library_read_type(self, library_read_type):
+        else:
+            new_val = self.get_new_mapping(field_name, mmap.values())
+            self.ref_map[metric_type][field_name] = new_val
+            self.session.execute(self.ref_table.insert(), [{'qc_program':self.qc_program, 'qc_metric':metric_type, 'field_name':field_name, 'field_code':new_val, 'display_name':field_name}])
```
Since `display_name` is just the `field_name`, shouldn't we delete it?

mcmcgrath13 replied:
This would be a manual database update so that something other than the field_name could be displayed on the front end.  I default-populated it with the field_name so it wasn't null.  I think this was a feature request from Ashok.

aguang commented on qcdb/parsers/parse.py:
```diff
@@ -50,10 +57,78 @@ def get_library_read_type(self, library_read_type):
+
+    # utility function to create field code
+    def get_new_mapping(self, field_name, codes):
+        print("Creating new code for ", field_name, " these codes already exist:", codes)
```
please use the logging module instead of print statements.

mcmcgrath13 replied:
all set

aguang reviewed (changes requested):
It looks good to me. Just a few mostly documentation comments:

 * `README.md` should have `docker-compose exec mysql mysql -p` now, otherwise you will get the error `ERROR: No such service: db`
 * `README.md` should also contain note that if you are loading db for the first time with new metrics or data that you need to use the `--buildref` flag, otherwise you will get an error.

mcmcgrath13 committed:
fix: update docs, fix logging

mcmcgrath13 commented:
@aguang latest push has your requested updates

mcmcgrath13 committed:
fix(parsing): if file errors, move to next instead of skipping all

#18

mcmcgrath13 committed:
fix(parsing): pass correct parset to correct module

aguang committed:
fix(db_load): close #18

fixed qckitfastq picardtools parser to correspond with skipping files adjustment made in db_load

aguang committed:
test(db_load): tests needed to be updated for new parse style
